cli/test/cli.ts

```diff
@@ -497,4 +501,73 @@ describe("CLI", () => {
                 done();
             });
     });
+
+    it("release doesn't allow releasing .zip file", (done: MochaDone): void => {
+        var command: cli.IReleaseCommand = {
+            type: cli.CommandType.release,
+            appName: "a",
+            deploymentName: "Staging",
+            description: "test releasing zip file",
+            mandatory: false,
+            appStoreVersion: "1.0.0",
+            package: "/fake/path/test/file.zip"
+        };
+
+        var release: Sinon.SinonSpy = sandbox.spy(cmdexec.sdk, "release");
+
+        cmdexec.execute(command)
+            .done((): void => {
+            }, (error: any): void => {
+                if (error) {
+                    assert.equal(error.message, "It is unnecessary to package releases in a .zip or binary file. Please specify the path of the desired directory or file directly.");
+                    done();
+                }
+            });
+    });
+
+    it("release doesn't allow releasing .ipa file", (done: MochaDone): void => {
+        var command: cli.IReleaseCommand = {
+            type: cli.CommandType.release,
+            appName: "a",
+            deploymentName: "Staging",
+            description: "test releasing zip file",
+            mandatory: false,
+            appStoreVersion: "1.0.0",
+            package: "/fake/path/test/file.ipa"
+        };
+
+        var release: Sinon.SinonSpy = sandbox.spy(cmdexec.sdk, "release");
+
+        cmdexec.execute(command)
+            .done((): void => {
+            }, (error: any): void => {
+                if (error) {
+                    assert.equal(error.message, "It is unnecessary to package releases in a .zip or binary file. Please specify the path of the desired directory or file directly.");
+                    done();
+                }
+            });
```

Comment:
const RELEASE_FAILED_ERROR_MESSAGE = "...";
    
    ...
    
    cmdexec.execute(command)
        .done((): void => throw Error("Error expected"), (error: any): void => {
            assert(!!error);
            assert.equal(error.message, RELEASE_FAILED_ERROR_MESSAGE);
            done();
        });

This way the error message is deduplicated, the done() resolved and rejected-with-no-error code paths throw a clear error (instead of timing out).  You could also deduplicate the `command` initialization with a helper function.